plugins/guests/linux/cap/halt.rb

```diff
@@ -1,10 +1,18 @@
+require 'vagrant/util/guest_inspection'
+
 module VagrantPlugins
   module GuestLinux
     module Cap
       class Halt
+        extend Vagrant::Util::GuestInspection::Linux
+
         def self.halt(machine)
           begin
-            machine.communicate.sudo("shutdown -h now")
+            if systemd?(machine.communicate)
+              machine.communicate.sudo("systemctl halt")
```

Comment:
I think this should use `systemctl poweroff` because `halt` leaves the machine running.